JavaScript/chapter01/1.6 - String Compression/rroque98_sol.js

```diff
@@ -8,9 +10,9 @@ const stringCompression = (str) => {
     return '';
   }
   var compStr = '';
-  var count = 1;
+  var count = 0;
   var currentChar = str[0];
-  for (let i = 1; i < str.length; i++) {
+  for (let i = 0; i < str.length; i++) {
     let char = str[i];
     if (char === currentChar) {
       count++;
```

Comment:
I cannot add a comment on L19, but this is the well-known [for-if antipattern](https://blogs.msdn.microsoft.com/oldnewthing/20111227-00/?p=8793).

Do you think you can restructure the code in a way in which you avoid checking `if (i === str.length - 1)`?